File: emailer/mail-service/src/main/java/io/emailer/entity/Mail.java
package io.emailer.entity;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "mail")
public class Mail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String fromName;
    private String fromEmail;
    private String toName;
    private String toEmail;
    private String subject;
    private String message;
    @Enumerated(EnumType.STRING)
    private MailStatus status;
    @ManyToOne
    private MailProvider mailProvider;
}

File: frontend/src/app/model/mail.ts
export interface Mail {
  fromName: string;
  fromEmail: string;
  toName: string;
  toEmail: string;
  subject: string;
  message: string;
}

export interface MailDetails {
  id: number;
  fromName: string;
  fromEmail: string;
  toName: string;
  toEmail: string;
  subject: string;
  message: string;
  status: string;
  mailProvider?: string;
}